test/terra/backends/qasm_simulator/qasm_snapshot.py

```diff
@@ -0,0 +1,338 @@
+        statevec_targets = snapshot_state_post_measure_statevector_deterministic(
+        )
+        circuits = snapshot_state_circuits_deterministic(label,
+                                                         'stabilizer',
+                                                         post_measure=True)
+
+        qobj = assemble(circuits, self.SIMULATOR, memory=True, shots=shots)
+        job = self.SIMULATOR.run(qobj, backend_options=self.BACKEND_OPTS)
+        method = self.BACKEND_OPTS.get('method', 'automatic')
+        if method not in QasmSnapshotStabilizerTests.SUPPORTED_QASM_METHODS:
+            self.assertRaises(AerError, job.result)
+        else:
+            result = job.result()
+            self.is_completed(result)
+            self.compare_counts(result, circuits, counts_targets, delta=0)
+            # Check snapshots
+            for i, circuit in enumerate(circuits):
+                data = result.data(circuit)
+                snaps = self.stabilizer_snapshots(data, label)
+                for j, mem in enumerate(data['memory']):
+                    statevec = statevec_targets[i].get(mem)
+                    stabilizer = snaps[j]
+                    self.assertTrue(
+                        self.stabilizes_statevector(stabilizer, statevec))
+
+    def test_snapshot_stabilizer_post_measure_nondet(self):
+        """Test snapshot stabilizer after non-deterministic final measurement"""
+        shots = 1000
+        label = "snap"
+        counts_targets = snapshot_state_counts_nondeterministic(shots)
+        statevec_targets = snapshot_state_post_measure_statevector_nondeterministic(
+        )
+        circuits = snapshot_state_circuits_nondeterministic(label,
+                                                            'stabilizer',
+                                                            post_measure=True)
+
+        qobj = assemble(circuits, self.SIMULATOR, memory=True, shots=shots)
+        job = self.SIMULATOR.run(qobj, backend_options=self.BACKEND_OPTS)
+        method = self.BACKEND_OPTS.get('method', 'automatic')
+        if method not in QasmSnapshotStabilizerTests.SUPPORTED_QASM_METHODS:
+            self.assertRaises(AerError, job.result)
+        else:
+            result = job.result()
+            self.is_completed(result)
+            self.compare_counts(result,
+                                circuits,
+                                counts_targets,
+                                delta=0.1 * shots)
+            # Check snapshots
+            for i, circuit in enumerate(circuits):
+                data = result.data(circuit)
+                snaps = self.stabilizer_snapshots(data, label)
+                for j, mem in enumerate(data['memory']):
+                    statevec = statevec_targets[i].get(mem)
+                    stabilizer = snaps[j]
+                    if not self.stabilizes_statevector(stabilizer, statevec):
+                        print("\n", circuit.draw())
```

Comment:
We should be using the `logger` instead of `print`

Reply:
Whoops, this shouldnt be there...